src/test/java/org/openelisglobal/provider/ProviderServiceTest.java

```diff
@@ -68,4 +79,57 @@ public void getProviderByFhirUuid_shouldReturnTheProviderGivenFhirUUID() {
         assertEquals("provider1 should have an provider type G", "G", provider1.getProviderType());
         assertEquals("provider2 should have an provider type s", "S", provider2.getProviderType());
     }
+
+    @Test
+    public void getProviderPages_shouldReturnPagesWithProviders() {
+        List<Provider> providersPage = providerService.getPageOfProviders(1);
+        int expectedPages = Integer
+                .parseInt(ConfigurationProperties.getInstance().getPropertyValue("page.defaultPageSize"));
+        assertTrue(providersPage.size() <= expectedPages);
+    }
+
+    @Test
+    public void getPagesOfSearchedProviders_shouldReturnPagesOfSearchedProvider() {
+        Provider provider1 = providerService.getProviderByFhirId(FH_UUID1);
+        String providerType = provider1.getProviderType();
+        List<Provider> providersSearched = providerService.getPagesOfSearchedProviders(1, providerType);
+        int expectedPages = Integer
+                .parseInt(ConfigurationProperties.getInstance().getPropertyValue("page.defaultPageSize"));
+        assertTrue(providersSearched.size() <= expectedPages);
+
+    }
+
+    @Test
+    public void deactivateProviders_shouldSetAllActiveFalse() {
+        List<Provider> providers = providerService.getAllProviders();
+        providerService.deactivateProviders(providers);
+        List<Provider> providersDeactivated = providerService.getAllProviders();
+        providersDeactivated.forEach(provider -> assertFalse("should have false on active", provider.getActive()));
+
+    }
+
+    @Test
+    public void insertOrUpdateProviderByFhirUuid_shouldUpdateProviderGivenTheFhirUUID() {
+        Provider provider1 = new Provider();
+        Person person1 = new Person();
```

Comment:
you can use datasets here aswell by adding person record in the dataset `provider.xml` and  then fetch it by `personservice.get(personid);`. Then use it to set person for the provider.